src/main/kotlin/graphics/scenery/utils/Image.kt

```diff
@@ -54,36 +72,134 @@ open class Image(val contents: ByteBuffer, val width: Int, val height: Int, val
+                    logger.debug("Display window : " + header.display_window().min_x() + ", " + header.display_window().min_y() + " -> " + header.display_window().max_x() + ", " + header.display_window().max_y())
+                    logger.debug("Screen window center : " + header.screen_window_center()[0] + ", " + header.screen_window_center()[1])
+                    logger.debug("Screen window width : " + header.screen_window_width())
+                    logger.debug("Chunk count : " + header.chunk_count())
+                    logger.debug("Tiled : " + header.tiled())
+                    logger.debug("Tile size x : " + header.tile_size_x())
+                    logger.debug("Tile size y : " + header.tile_size_y())
+                    logger.debug("Tile level mode : " + header.tile_level_mode())
+                    logger.debug("Tile rounding mode : " + header.tile_rounding_mode())
+                    logger.debug("Long name : " + header.long_name())
+                    logger.debug("Non-image : " + header.non_image())
+                    logger.debug("Multipart : " + header.multipart())
+                    logger.debug("Header length : " + header.header_len())
+                    logger.debug("Number of custom attributes : " + header.num_custom_attributes())
+                    logger.debug("Custom attributes : " + header.custom_attributes())
+                    logger.debug("Channels : " + header.channels().remaining())
+                    logger.debug("Pixel types : " + header.pixel_types().remaining())
+                    logger.debug("Number of channels : " + header.num_channels())
+                    logger.debug("Compression type : " + header.compression_type())
+                    logger.debug("Requested pixel types : " + header.requested_pixel_types().remaining())
+                    logger.debug("Name : " + header.nameString())
+
+                    image = EXRImage.malloc(stack)
+                    TinyEXR.InitEXRImage(image)
+                    result = TinyEXR.LoadEXRImageFromMemory(image, header, memory, err)
+                    check(result, err)
+                    logger.info("Parsed EXR image successfully.")
+                    logger.debug("------------------------------")
+                    logger.debug("Level x: " + image.level_x())
+                    logger.debug("Level y: " + image.level_y())
+                    logger.debug("Images: " + image.images()!!.remaining())
+                    logger.debug("Width: " + image.width())
+                    logger.debug("Height: " + image.height())
+                    logger.debug("Number of channels: " + image.num_channels())
+                    logger.debug("Number of tiles: " + image.num_tiles())
+
+                    //FreeEXRImage(image)
+                    TinyEXR.FreeEXRHeader(header)
+
+                    val pb = image.images()!!
```

Comment:
There should be a check here if `images()` actually has a non-null return value. Scanline images would not have iirc. Please issue a warning or exception in that case.